packages/sitecore-jss/src/site/graphql-siteinfo-service.ts

```diff
@@ -0,0 +1,137 @@
+import { GraphQLClient, GraphQLRequestClient } from '../graphql';
+import debug from 'debug';
+import { CacheClient, CacheOptions, MemoryCacheClient } from '../cache-client';
+
+const siteGroupingTemplate = 'EDA823FC-BC7E-4EF6-B498-CD09EC6FDAEF';
+
+const defaultQuery = /* GraphQL */ `
+  {
+    search(
+      where: { name: "_template", value: "${siteGroupingTemplate}", operator: EQ }
+    ) {
+      results {
+        ... on SiteDefinition {
+          name
+          hostName {
+            value
+          }
+          virtualFolder {
+            value
+          }
+          language: language_f19277fe1b854b0a8c265e74d766b3a3 {
+            value
+          }
+        }
+      }
+    }
+  }
+`;
+
+export type SiteInfo = {
+  name: string;
+  hostName: string;
+  virtualFolder: string;
+  language: string;
+};
+
+type GraphQLSiteInfoServiceOptions = {
+  /**
+   * Your Graphql endpoint
+   */
+  endpoint: string;
+  /**
+   * The API key to use for authentication
+   */
+  apiKey: string;
+  /**
+   * Settings for internal cache.
+   * Cache is enabled by default with timout set to 10
+   */
+  cacheSettings?: CacheOptions;
```

Comment:
Just a reminder to match pending decision here: https://github.com/Sitecore/jss/pull/1218#discussion_r1011859187